source/options.tsx

```diff
@@ -10,6 +10,59 @@ import * as indentTextarea from 'indent-textarea';
 
 import {perDomainOptions} from './options-storage';
 
+function setValidationStatus(text: string): void {
+	select('#validation')!.textContent = text;
+	select('form ul')!.classList.toggle('invalid', !text.includes('✔️'));
+}
+
+async function getHeaders(personalToken: string): Promise<string> {
```

Comment:
Make function names reflect what they do. This isn't a plain getter. This function should be refactored

Reply:
Is it good now?

Not sure how to spit it any more since I need to get the response if it errors

Reply:
Nothing changed, you just moved some code into a separate function, but the function is still being called from here.

- This doesn't return multiple headers, so it shouldn't be called headerS
- This is returning scopes, so return scopes, pre-parsed into an array


```suggestion
async function getTokenScopes(personalToken: string): Promise<string[]> {
```